src/passwordcheck.js

```diff
@@ -28,29 +36,26 @@ function ($, bloomfilter) {
     return regex.test(password);
   }
 
-  return function checkPasswordStrength(password, callback) {
-    if (password.length < 8) {
-      return;
-    }
-    var bool = false;
-    var strong = strengthCheck(password.toString());
-
-    if (strong) {
-      // Only if the password has a chance of being strong do we check with the bloom filter
-      // else, simply reject the password. This helps us to not store all-alpha or all-numeric passwords
-      // on the bloom filter, reducing space.
-      bool = bloomfilter.test(password);
-      callback('Password strength check with bloomfilter triggered');
-    } else {
-      bool = true;
-      callback('Password with all alphabets/numbers detected');
-    }
+  return function (password, callback) {
+    if (password && typeof password === 'string' && password.length > 8) {
```

Comment:
Addendum, maybe we could use a few more error codes here for completeness.

If there is no password, `PASSWORD_NOT_GIVEN`
If the password is not a string, `PASSWORD_NOT_A_STRING`